swarm/storage/mru/resource.go

```diff
@@ -274,47 +235,43 @@ func NewHandler(params *HandlerParams) (*Handler, error) {
 	return rh, nil
 }
 
-// SetStore sets the store backend for resource updates
+// SetStore sets the store backend for the Mutable Resource API
 func (h *Handler) SetStore(store *storage.NetStore) {
 	h.chunkStore = store
 }
 
-// Validate is a chunk validation method (matches ChunkValidatorFunc signature)
-//
-// If resource update, owner is checked against ENS record of resource name inferred from chunk data
-// If parsed signature is nil, validates automatically
-// If not resource update, it validates are root chunk if length is metadataChunkOffsetSize and first two bytes are 0
+// Validate is a chunk validation method
+// If it's a resource update, the chunk address is checked against the public key of the update's signature
+// It implements the storage.ChunkValidator interface
 func (h *Handler) Validate(addr storage.Address, data []byte) bool {
+
+	// does the chunk data make sense?
+	// This is not 100% safe evaluation, since content addressed data can coincidentally produce data with length header matching content size
 	signature, period, version, name, parseddata, _, err := h.parseUpdate(data)
 	if err != nil {
-		log.Warn(err.Error())
-		if len(data) > metadataChunkOffsetSize { // identifier comes after this byte range, and must be at least one byte
-			if bytes.Equal(data[:2], []byte{0, 0}) {
-				return true
-			}
-		}
-		log.Error("Invalid resource chunk")
+		log.Warn("Invalid resource chunk")
 		return false
-	} else if signature == nil {
-		return bytes.Equal(h.resourceHash(period, version, ens.EnsNode(name)), addr)
 	}
 
+	// Check if signature is valid against the chunk address
+	// Since we force signatures to be used, we can know be sure that this is valid data
 	digest := h.keyDataHash(addr, parseddata)
-	addrSig, err := getAddressFromDataSig(digest, *signature)
+	publicKey, err := crypto.SigToPub(digest.Bytes(), signature[:])
 	if err != nil {
-		log.Error("Invalid signature on resource chunk")
+		log.Warn("Unparseable signature in resource chunk %s", addr)
 		return false
 	}
-	ok, _ := h.checkAccess(name, addrSig)
-	return ok
-}
-
-// If no ens client is supplied, resource updates are not validated
-func (h *Handler) IsValidated() bool {
-	return h.ownerValidator != nil
+	signAddr := crypto.PubkeyToAddress(*publicKey)
+	nameHash := ens.EnsNode(name)
+	checkAddr := h.resourceHash(period, version, nameHash, signAddr)
+	if !bytes.Equal(checkAddr, addr) {
+		log.Warn("Invalid signature on resource chunk")
```

Comment:
We moved the logging of chunk validation to localstore put method, quite recently. Should this log still be here @zelig?

Reply:
It's hard to keep track of this stuff with incessant conflicts piling up over and over again :/ I think I'd prefer dedicated PRs for linting like this in the future.